### PR TITLE
Mark kubelet structured logs as migrated

### LINKED ISSUE #98976: Kubelet migration to structured logs
For Structured Logging Migration: https://github.com/kubernetes/enhancements/issues/1602

[Beta graduation criteria:](https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging#beta)
- Structured logging interface design is verify by migrating one whole component (Kubelet) to structured logging. All other component migrations will be best effort.

### What would you like to be added:

Please migrate large chunks of code in each PR, at **minimum** one top level directory at a time (assuming your diff does not exceed ~40 lines). There are already ~40 top-level directories in the kubelet. If we have fewer PRs, it will be much easier for reviewers and approvers to merge your change.

The following query can be used to identify directories that still need structured logging migration:

```bash
git grep -L klog pkg/kubelet/ | sed 's|[^/]\+$||' | sort -u
git grep -L klog cmd/kubelet/ | sed 's|[^/]\+$||' | sort -u
```

You can check the project board to see if someone has already migrated a certain file: https://github.com/orgs/kubernetes/projects/53

#### Authors: What to include in your PR

- **Subject:** "Migrate `<directory>` to structured logging"
- **SIGs:** instrumentation, node
- **Priority:** important-soon
- **Kind:** cleanup
- **Release note:** You **must** include a release note for any log changes!

**HOWTO:** Read the [structured logging migration guide](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md).

_Only_ reference this issue for kubelet migrations. Don't mark it as "fixes" or other keywords it will close this tracker. All other components are best effort for 1.21 and should have priority/backlog.

Before you submit your PR, check to see if someone has already migrated that component: https://github.com/kubernetes/kubernetes/pulls?q=is%3Apr+is%3Aopen+structured+log+label%3Asig%2Fnode

#### Reviewers: What to look for

Read through the migration guide above. Use the PR query above to find PRs to review. Follow standard [node triage guidelines](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-node/triage.md).

### Why is this needed:

See above.

/sig instrumentation
/sig node
/cc @serathius 
/milestone v1.21

### PR DESCRIPTION
**Pull request by ehashman**
#### What type of PR is this?

/kind cleanup
/sig instrumentation

#### What this PR does / why we need it:

Verifying what remains to be migrated for kubelet structured logs

#### Which issue(s) this PR fixes:

Fixes #98976.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by ehashman** [comment]
/priority important-soon
/triage accepted
/milestone v1.21

Hopefully the last one x_x

**Comment by ehashman** [comment]
Note: this change is test only, with the exception of any missed log line entries I will need to go back and fix once [all structured logging PRs merge](https://github.com/orgs/kubernetes/projects/53).

**Comment by pacoxu** [comment]
/retest
/skip
as all are merged.

**Comment by pacoxu** [comment]
It seems there are several remaining failures. 
Let me fix it.

File: pkg/kubelet/prober/prober.go [hunk 1]
```diff
@@ -192,7 +192,7 @@ func (pb *prober) runProbe(probeType probeType, p *v1.Probe, pod *v1.Pod, status
 		if host == "" {
 			host = status.PodIP
 		}
-		klog.V(4).Infof("TCP-Probe Host: %v, Port: %v, Timeout: %v", host, port, timeout)
+		klog.V(4).InfoS("TCP-Probe Host", "host", host, "port", port, "timeout", timeout)
 		return pb.tcp.Probe(host, port, timeout)
 	}
 	klog.Warningf("Failed to find probe builder for container: %v", container)
```
Comment (pacoxu) [inline pkg/kubelet/prober/prober.go:198]: Warningf here

**Comment by ehashman** [comment]
Pulled @pacoxu's changes into my PR, let's see if this passes verification.

**Comment by ehashman** [comment]
/skip
flaky test #99911

/test pull-kubernetes-unit

**Comment by ehashman** [comment]
/cc @BenTheElder 
for hack/

/cc @serathius
for instrumentation
 
/cc @mrunalp 
for node

**Comment by BenTheElder** [comment]
/approve
for hack/

File: pkg/kubelet/prober/prober.go [hunk 2]
```diff
@@ -192,10 +192,10 @@ func (pb *prober) runProbe(probeType probeType, p *v1.Probe, pod *v1.Pod, status
 		if host == "" {
 			host = status.PodIP
 		}
-		klog.V(4).Infof("TCP-Probe Host: %v, Port: %v, Timeout: %v", host, port, timeout)
+		klog.V(4).InfoS("TCP-Probe Host", "host", host, "port", port, "timeout", timeout)
 		return pb.tcp.Probe(host, port, timeout)
 	}
-	klog.Warningf("Failed to find probe builder for container: %v", container)
+	klog.InfoS("Failed to find probe builder for container", "container", container.Name)
```
Comment (serathius) [inline pkg/kubelet/prober/prober.go:198]:
```suggestion
	klog.InfoS("Failed to find probe builder for container", "containerName", container.Name)
```

File: pkg/kubelet/kubelet_volumes.go [hunk 3]
```diff
@@ -153,7 +153,7 @@ func (kl *Kubelet) cleanupOrphanedPodDirs(pods []*v1.Pod, runningPods []*kubecon
 					orphanVolumeErrors = append(orphanVolumeErrors, fmt.Errorf("orphaned pod %q found, but failed to rmdir() subpath at path %v: %v", uid, subpathVolumePath, err))
 					allVolumesCleanedUp = false
 				} else {
-					klog.Warningf("Cleaned up orphaned volume subpath from pod %q at %s", uid, subpathVolumePath)
+					klog.InfoS("Cleaned up orphaned volume subpath from pod", "podUID", uid, "subpathVolumePath", subpathVolumePath)
```
Comment (serathius) [inline pkg/kubelet/kubelet_volumes.go:156]:
```suggestion
					klog.InfoS("Cleaned up orphaned volume subpath from pod", "podUID", uid, "path", subpathVolumePath)
```

File: pkg/kubelet/kubelet_volumes.go [hunk 4]
```diff
@@ -136,7 +136,7 @@ func (kl *Kubelet) cleanupOrphanedPodDirs(pods []*v1.Pod, runningPods []*kubecon
 					orphanVolumeErrors = append(orphanVolumeErrors, fmt.Errorf("orphaned pod %q found, but failed to rmdir() volume at path %v: %v", uid, volumePath, err))
 					allVolumesCleanedUp = false
 				} else {
-					klog.Warningf("Cleaned up orphaned volume from pod %q at %s", uid, volumePath)
+					klog.InfoS("Cleaned up orphaned volume from pod", "podUID", uid, "volumePath", volumePath)
```
Comment (serathius) [inline pkg/kubelet/kubelet_volumes.go:139]:
```suggestion
					klog.InfoS("Cleaned up orphaned volume from pod", "podUID", uid, "path", volumePath)
```

File: pkg/kubelet/prober/prober.go (hunk 2 above)
Comment (ehashman) [reply pkg/kubelet/prober/prober.go]: Found a bunch of others in the file, fixed those too.

**Review by kikisdeliveryservice — COMMENTED** [review]
in: kubelet_volumes.go line 70:
```klog.V(4).InfoS("Pod found, but volumes are still mounted on disk", "podUID", podUID, "volumePaths", volumePaths)```

volumePaths -> paths?

**Comment by ehashman** [comment]
https://github.com/kubernetes/kubernetes/blob/b57a8bacaddc839e6edc6757c25897435782439e/pkg/kubelet/cm/cgroup_manager_linux.go#L295

We do use `paths` elsewhere so may as well

**Pull request merged**